flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetTableSource.java

```diff
@@ -447,8 +453,16 @@ private String getColumnName(BinaryComparison comp) {
 
 	@Nullable
 	private Tuple2<Column, Comparable> extractColumnAndLiteral(BinaryComparison comp) {
-		TypeInformation<?> typeInfo = getLiteralType(comp);
 		String columnName = getColumnName(comp);
+		TypeInformation<?> typeInfo = null;
+		try {
+			String[] columnPath = columnName.split("\\.");
```

Comment:
You can have a `ColumnPath columnPath = ColumnPath.fromDotString(columnName);`
And use `parquetSchema.getType(columnPath.toArray())`.
In this way, you can avoid introduce hard code `columnName.split("\\.")`.

Reply:
Thanks for the suggestion. Changed.